implementation/test/DepositRedemptionTest.js

```diff
@@ -743,6 +743,15 @@ describe("DepositRedemption", async function() {
         "Invalid signature",
       )
     })
+
+    it("reverts if S value is on the upper half of the secp256k1 curve's order", async () => {
+      const s =
+        "0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF5D576E7357A4501DDFE92F46681B20A1"
```

Comment:
Just a note - a better unit test would take a valid signature, flip the 's' value, check that the signature passes `ecrecover`, then submit to `provideRedemptionSignature` and expect the "Malleable signature" failure.